app/src/main/java/com/gatchimath/totask/TaskStorage.java

```diff
@@ -0,0 +1,81 @@
+package com.gatchimath.totask;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class represents task-related storage in the app.
+ * It's purpose is to broker between other classes and
+ * storage classes, like app storage or cloud storage.
+ * The rest of the app does not see and should not care
+ * where data goes and comes from, it should only see and
+ * care about TaskStorage.  Likewise, only TaskStorage gets to
+ * care where things go and come from.
+ *
+ * However, because Tasks [will soon be] Observables, TaskStorages should
+ * get very little interaction outside of its world.  It's mainly
+ * for adding and removing tasks from storage.
+ *
+ * @author  gatchi (github.com/gatchi)
+ */
+public class TaskStorage {
+	
+	private ArrayList<Task> taskList;
+	
+	public TaskStorage() {
+		taskList = new ArrayList();
+	}
+
+	/**
+	 * Adds the task to this storage object.
+	 *
+	 * @param task  The task to pass into this storage.
+	 */
+	public void addTask(Task task) {
+		taskList.add(task);
+		syncStorage();
```

Comment:
A one-size fits all syncStorage method probably won't be applicable, since
a) We can't add the task until we find out whether persisting to storage succeeded or not and
b) Each type of operation will need to be implemented separately to be efficient.

Accordingly, you can probably leave this out for now, and we can add in methods later to attach an AppStorage to this class.